### DISCUSSION TITLE
Community call December

### DISCUSSION THREAD

**Discussion by eddiejaoude** [EddieHubCommunity/support#3087] [Tips and Tricks]
Topic: Cloud development environments
Platform: Discord stages
Guest: @pawlean

Topics
- What is ephemeral dev environments
- Why use ephemeral dev environments
- Benefits of using Gitpod
- Why you should add it to your projects
- How to get started
- … other topics?
-Q&A throughout the discussion

Learn more at [Gitpod](https://eddiejaoude.krtra.com/t/91EQ8BvNXwa4) (plus it is [open source](https://github.com/gitpod-io)!)

Gitpod [videos](https://www.youtube.com/watch?v=VYHsd1zI_tE&list=PL3TSF5whlprXVp-7Br2oKwQgU4bji1S7H&index=4)

**Comment by eddiejaoude** [comment]
Q. Have you used a cloud development environment before?

*For example: CodeSpaces, Gitpod or ...*

**Reply by suitupstan** [reply]
I've used GitPod, CodeSpaces, VSCode.dev, CodeSandBox and CodePen before

**Reply by naomi-lgbt** [reply]
GitPod and CodePen here. Little bit of CodeSpaces but not much.

**Reply by Pradumnasaraf** [reply]
I've used GitPod, VSCode.dev

**Comment by eddiejaoude** [comment]
Q. Would you use a cloud development environment?

*If no, why not?*

**Reply by suitupstan** [reply]
I would not be using a cloud development environment for things I consider as "complex dev environments".

I would use a CDE for all other things, like tiny itsy bitsy contributions and more 🙂

**Reply by Cahllagerfeld** [reply]
Not using anything else anymore 👀

**Reply by naomi-lgbt** [reply]
I develop locally, because I've got a machine that can handle it.

If things change in a way that I start travelling, I might migrate to a cloud environment just for the remote access ability.

**Reply by Pradumnasaraf** [reply]
Ya if I have to contribute a project, and that needs X number of variables to set up.

**Reply by mikeysan** [reply]
I already use one, GitPod. However, I hesitate using it all the time as I am yet to learn (prior to today) how to set it up to have my python and virtual python environment all the time.

**Comment by eddiejaoude** [comment]
Q. What is ephemeral to you?

**Reply by naomi-lgbt** [reply]
Temporary.

**Reply by suitupstan** [reply]
Lasting for a limited time

**Comment by naomi-lgbt** [comment]
Don't forget to drop your favourite things you learned in today's call below! 👇🏻

**Reply by naomi-lgbt** [reply]
@eddiejaoude learned how to say `ephemeral` 😁

**Comment by suitupstan** [comment]
### What is ephemeral dev environments?

My understanding of the English language tells me that they are "**temporary**" or short time environments, as the word ephemeral means.

Since I've used gitpod before I can conclude that "ephemeral dev environments" are **"A development environment which lasts for a limited amount of time and is sort of containerized, if I talk in docker language"**

### Why use ephemeral dev environments?

If you are gonna work on it just for a temporary amount of time, it is not worth setting up the environment, making a slight change and never using that environment again 🤷 It saves you time, makes you more productive and helps you in an infinitive amount of ways.

### Benefits of using Gitpod

- [x] Quick Environment Setup
- [x] Never face "your environment doesn't support this"
- [x] Be more productive
- [x] Remote development
- [x] Your dev env === prod env
- [x] Code on tablet? 😆

### Why you should add it to your projects?

- [x] Encourages contributions
- [x] Eases the maintenance process (reviewing PRs)
- [x] All the benefits of course
- [x] Quick Testing env

### How to get started?

- For projects => https://www.gitpod.io/blog/gitpodify
- For contributors => https://www.gitpod.io/#get-started

**Comment by mikeysan** [comment]
I have used GitPod briefly and got excited about vscode.dev but have not tried that yet.

Q: Can I have a consistent GitPod workspace environment?

**Reply by suitupstan** [reply]
Yeah, you can customize your gitpod environment it will be global... It will also sync with your codespaces, local vscode, and vscode.dev with the vscode settings sync feature. This means consistent env everywhere 😄

**Comment by suitupstan** [comment]
My questions 👇

**Reply by suitupstan** [reply]
Some extensions don't work on gitpod for some reason, I tried searching but didn't find them, they are on local though

**Reply by suitupstan** [reply]
I tried running a app that uses GitHub Oauth and the app didn't work, while it works on local and prod

**Comment by naomi-lgbt** [comment]
Here's resources so they don't get lost in chat:

Extension Marketplace that GitPod uses: https://open-vsx.org/
GitPod YouTube channel: https://www.youtube.com/c/gitpod

**Reply by suitupstan** [reply]
Thanks for sharing ❤️

**Comment by eddiejaoude** [comment]
Apply for free open source account https://www.gitpod.io/docs/professional-open-source#who-is-eligible

**Comment by pawlean** [comment]
🧡 https://www.gitpod.io/community/heroes

**Comment by pawlean** [comment]
Building Gitpod in Gitpod: https://www.youtube.com/watch?v=TIT4FvZe6fY

**Comment by BOLT04** [comment]
Hi everyone 👋 , does anyone know if Gitpod (or cloud dev environments in general) support offline capabilities? So let's say someone travels a lot or goes to places where there is a slow connection or there is really no internet. Could we access our environments offline 🤔 ?

I'm mostly curious if this is a concern for other people or not, it might not be a feature that was implemented because it's not necessary 🙂.

**Reply by mikeysan** [reply]
I think @eddiejaoude asked something similar during the call session. I believe it was intriguing enough that @pawlean said they would try it out and get back to us.

My take is that this is not possible (at least right now).  

FYI: Eddie's scenario was losing connection while airborne (or something like that) :)

**Reply by BOLT04** [reply]
awesomeee, thanks @mikeysan 👍. I had to leave the call but it's really cool that this topic was discussed and is being investigated by the Gitpod team 😃
